dashboard/src/reducers/auth.ts

```diff
@@ -24,17 +26,22 @@ const authReducer = (state: IAuthState = initialState, action: AuthAction): IAut
         authenticated: action.payload.authenticated,
         oidcAuthenticated: action.payload.oidc,
         authenticating: false,
+        // If authenticated the session is no longer expired
+        sessionExpired: action.payload.authenticated ? false : state.sessionExpired,
       };
     case getType(actions.auth.authenticating):
       return { ...state, authenticated: false, authenticating: true };
     case getType(actions.auth.authenticationError):
       return {
         ...state,
-        authenticated: false,
-        authenticating: false,
-        oidcAuthenticated: false,
         authenticationError: action.payload,
       };
+    case getType(actions.auth.setExpiredSession):
+      return {
+        ...state,
+        // The session can only expire if we are using OIDC tokens
+        sessionExpired: state.oidcAuthenticated ? true : state.sessionExpired,
```

Comment:
We should move this check to the action, the reducer should be simpler and just set the field to true always if the action has been dispatched.